stdlib/src/utils/inline_string.mojo

```diff
@@ -147,28 +147,15 @@ struct InlineString(Sized, Stringable, CollectionElement, CollectionElementNew):
             # Begin by heap allocating enough space to store the combined
             # string.
             var buffer = List[UInt8](capacity=total_len)
-
             # Copy the bytes from the current small string layout
-            memcpy(
-                dest=buffer.unsafe_ptr(),
-                src=self._storage[_FixedString[Self.SMALL_CAP]].unsafe_ptr(),
-                count=len(self),
+            var span_self = Span[Byte, __origin_of(self)](
+                ptr=self._storage[_FixedString[Self.SMALL_CAP]].unsafe_ptr(),
+                length=len(self),
             )
-
+            buffer.append(span_self)
             # Copy the bytes from the additional string.
-            memcpy(
-                dest=buffer.unsafe_ptr() + len(self),
-                src=str_slice.unsafe_ptr(),
-                count=str_slice.byte_length(),
-            )
-
-            # Record that we've initialized `total_len` count of elements
-            # in `buffer`
-            buffer.size = total_len
-
-            # Add the NUL byte
-            buffer.append(0)
-
+            buffer.append(str_slice.as_bytes())
+            buffer.append(0)  # Add the NUL byte
```

Comment:
Nice! This is a great cleanup. In theory this might be slightly less efficient since the `append()` will need to perform redundant bounds-checking. But I think that's a small cost that we shouldn't worry too much about optimizing yet.

Reply:
> In theory this might be slightly less efficient since the append() will need to perform redundant bounds-checking.

We could add a `unsafe_no_checks` parameter to all the append functions at some point. That way we'd stop having to resort to `UnsafePointer` so much for such scenarios.